nucliadb/src/nucliadb/search/api/v1/catalog.py

```diff
@@ -93,23 +99,32 @@ async def catalog_get(
     ),
     hidden: Optional[bool] = fastapi_query(SearchParamDefaults.hidden),
 ) -> Union[KnowledgeboxSearchResults, HTTPClientError]:
-    item = CatalogRequest(
-        query=query,
-        filters=filters,
-        faceted=faceted,
-        page_number=page_number,
-        page_size=page_size,
-        debug=debug,
-        with_status=with_status,
-        range_creation_start=range_creation_start,
-        range_creation_end=range_creation_end,
-        range_modification_start=range_modification_start,
-        range_modification_end=range_modification_end,
-        hidden=hidden,
-    )
-    if sort_field:
-        item.sort = SortOptions(field=sort_field, limit=sort_limit, order=sort_order)
-    return await catalog(kbid, item)
+    try:
+        expr = (
+            CatalogFilterExpression.model_validate_json(filter_expression) if filter_expression else None
+        )
+
+        item = CatalogRequest(
+            query=query,
+            filter_expression=expr,
+            filters=filters,
+            faceted=faceted,
+            page_number=page_number,
+            page_size=page_size,
+            debug=debug,
+            with_status=with_status,
+            range_creation_start=range_creation_start,
+            range_creation_end=range_creation_end,
+            range_modification_start=range_modification_start,
+            range_modification_end=range_modification_end,
+            hidden=hidden,
+        )
+        if sort_field:
+            item.sort = SortOptions(field=sort_field, limit=sort_limit, order=sort_order)
+        return await catalog(kbid, item)
+    except ValidationError as exc:
```

Comment:
Is the `return await catalog()` line ok to be inside this try catch? I wonder if it would be masking errors on the serialization of the response as if they were client errors.

Also, should the InvalidQueryError be catched aswell?

Reply:
`InvalidQueryError` is handled inside the `catalog` function. I reduced the try scope so it only catches the expr parsing.